ipfscluster_test.go

```diff
@@ -565,6 +567,29 @@ func TestClustersPeers(t *testing.T) {
 	}
 }
 
+func TestClustersPeersRetainOrder(t *testing.T) {
+	ctx := context.Background()
+	clusters, mock := createClusters(t)
+	defer shutdownClusters(t, clusters, mock)
+
+	delay()
+
+	j := rand.Intn(nClusters) // choose a random cluster peer
+	peers1, err := json.Marshal(clusters[j].Peers(ctx))
```

Comment:
If you had not made the changes and had written this test before, chances are it would have passed just fine.

The order depended on metrics. You should probably do WaitForMetrics() in between. And use different peers. And repeat the comparison for like 5 times.